src/LogEventProcessor.ts

```diff
@@ -74,15 +77,20 @@ export default class LogEventProcessor {
     if (this.queue.length === 0) {
       return Promise.resolve();
     }
-    const oldQueue = this.queue;
+    let eventsToSync: LogEvent[] = this.queue;
     this.queue = [];
+    const queuedEventsInAdapter = this.loggingAdapter?.getQueuedEvents();
+    if (queuedEventsInAdapter && queuedEventsInAdapter.length > 0) {
```

Comment:
why do we have a queue in memory, and then a separate set of events in the adapter?  We need one source of truth here.  I think for non php sdks, we can leave this in memory for now - there is a risk here with multiple sdk instances sharing the same log queue in redis or another db and having to worry about transactions to not lose events.  If they each manage their own queue of events in memory, then we wont have this issue

Reply:
I was thinking this could back up any events that could not be synced with the network. But by design, this does risk being out of sync with the in-memory queue.
But currently, in-memory queue does not support a backlog of "failed-to-sync" events so doesn't the risk of lost events  already exist?